18-minimal/README.md

```diff
@@ -0,0 +1,322 @@
+```
+$ podman run --env DEV_MODE=true DEBUG_PORT=5454 my-image-id
+```
+
+To run the container in production mode, run:
+```
+$ podman run --env DEV_MODE=false my-image-id
+```
+
+By default, `DEV_MODE` is set to `false`, and `DEBUG_PORT` is set to `5858`, however the `DEBUG_PORT` is only relevant if `DEV_MODE=true`.
+
+Hot deploy
+----------
+
+As part of development mode, this image supports hot deploy. If development mode is enabled, any souce code that is changed in the running container will be immediately reflected in the running nodejs application.
+
+### Using Podman's exec
+
+To change your source code in a running container, use Podman's [exec](https://github.com/containers/libpod) command:
+```
+$ podman exec -it <CONTAINER_ID> /bin/bash
+```
+
+After you [Podman exec](https://github.com/containers/libpod) into the running container, your current directory is set to `/opt/app-root/src`, where the source code for your application is located.
+
+### Using OpenShift's rsync
+
+If you have deployed the container to OpenShift, you can use [oc rsync](https://docs.openshift.org/latest/dev_guide/copy_files_to_container.html) to copy local files to a remote container running in an OpenShift pod.
+
+#### Warning:
+
+The default behaviour of the s2i-nodejs container image is to run the Node.js application using the command `npm start`. This runs the _start_ script in the _package.json_ file. In developer mode, the application is run using the command `nodemon`. The default behaviour of nodemon is to look for the _main_ attribute in the _package.json_ file, and execute that script. If the _main_ attribute doesn't appear in the _package.json_ file, it executes the _start_ script. So, in order to achieve some sort of uniform functionality between production and development modes, the user should remove the _main_ attribute.
+
+Below is an example _package.json_ file with the _main_ attribute and _start_ script marked appropriately:
+
+```json
+{
+    "name": "node-echo",
+    "version": "0.0.1",
+    "description": "node-echo",
+    "main": "example.js", <--- main attribute
+    "dependencies": {
+    },
+    "devDependencies": {
+        "nodemon": "*"
+    },
+    "engine": {
+        "node": "*",
+        "npm": "*"
+    },
+    "scripts": {
+        "dev": "nodemon --ignore node_modules/ server.js",
+        "start": "node server.js" <-- start script
+    },
+    "keywords": [
+        "Echo"
+    ],
+    "license": "",
+}
+```
+
+#### Note:
+`oc rsync` is only available in versions 3.1+ of OpenShift.
+
+
+See also
+--------
+Dockerfile and other sources are available on https://github.com/sclorg/s2i-nodejs-container.
+In that repository you also can find another versions of Node.js environment Dockerfiles.
+Dockerfile for CentOS is called `Dockerfile`, Dockerfile for RHEL7 is called `Dockerfile.rhel7`,
+for RHEL8 it's `Dockerfile.rhel8` and the Fedora Dockerfile is called Dockerfile.fedora.
```

Comment:
Sorry for nitpicking. Can you please add also sentences about CentOS Stream8 and CentOS Stream 9?

Like: Dockerfile for CentOS Stream 8 is called `Dockerfile.c8s` and Dockerfile for CentOS Stream 9 is called `Dockerfile.c9s`

Reply:
These are not going to get built for c8s or c9s however - both are excluded.

Reply:
If they are excluded then it does not make sense to have them in the repository. I don't follow, why have them here if they are not built, tested, and even pushed to quay.io? Do we expect to support them soon? I so, then skip my comments.

Reply:
We don't know yet whether we'll be asked to have also the minimal variant for the tech-preview, it might be requested or might not. I thought including the result of already done work is not that bad, despite it can stay unused (I wanted to test whether the minimal image based on c8s works, to know ahead that we might expect problems later).

Reply:
I thought we did not have c8s and c9s minimal base images. Which why they are not excluded. Is that not the case?